addon/plugins/remove-properties-of-deleted-nodes/index.ts

```diff
@@ -0,0 +1,131 @@
+import { EditorState, Plugin } from 'prosemirror-state';
+import { PNode } from '@lblod/ember-rdfa-editor';
+import { getNodesByResource } from '@lblod/ember-rdfa-editor/plugins/rdfa-info';
+import { isResourceNode } from '@lblod/ember-rdfa-editor/utils/_private/node-utils';
+import {
+  getBacklinks,
+  getProperties,
+  getRdfaId,
+} from '@lblod/ember-rdfa-editor/utils/_private/rdfa-utils';
+import { removeBacklinksTargetingResourceProperty } from '@lblod/ember-rdfa-editor/commands/_private/rdfa-commands/remove-backlinks-targeting-resource-property';
+import { removePropertiesTargetingBacklinks } from '@lblod/ember-rdfa-editor/commands/_private/rdfa-commands/remove-properties-targeting-backlinks';
+
+/**
+ * Returns nodes from the oldState that are not in the newState
+ */
+function getDeletedNodes(oldState: EditorState, newState: EditorState) {
+  const prevNodesById: Map<string, PNode> = new Map();
+  const nextNodesById: Map<string, PNode> = new Map();
+
+  oldState.doc.descendants((node) => {
+    if (node.attrs['__rdfaId']) {
+      prevNodesById.set(node.attrs['__rdfaId'] as string, node);
+    }
+  });
+
+  newState.doc.descendants((node) => {
+    if (node.attrs['__rdfaId']) {
+      nextNodesById.set(node.attrs['__rdfaId'] as string, node);
+    }
+  });
```

Comment:
These two mappings are (I think) often calculated twice: as the `newState.doc` becomes the `oldState.doc` in the next iteration. We might be able to cache the mappings?